src/compile/mark/mark.ts

```diff
@@ -239,8 +239,8 @@ function getStackByFields(model: UnitModel) {
         const fieldDef: FieldDef = channelEncoding;
         const scale = model.scale(channel);
         fields.push(field(fieldDef, {
-          binSuffix: scale && scale.type === ScaleType.ORDINAL ? '_range' : '_start'
-        }));
+          binSuffix: scale && scale.type === ScaleType.ORDINAL ? 'range' : 'start'
+        }) || 'undefined');
```

Comment:
@kanitw @domoritz without this addition, a test fails because the vega renders to 
`groupby: [null]`
when it used to be
`groupby: ["undefined"]`

So I did this... I'm not really sure why getting rid of the _ here and adding it in field causes the value to become null. So... this passes all tests, but it might not be foolproof because I don't know why this happened. Please check it out before merging.